api/src/main/java/javax/enterprise/inject/spi/container/Container.java

```diff
@@ -0,0 +1,341 @@
+
+    /**
+     * Define the set of bean classes for the synthetic bean archive.
+     *
+     * @param classes classes to add to the synthetic bean archive
+     * @return self
+     */
+    Container beanClasses(Class<?>... classes);
+
+    /**
+     * Add a bean class to the set of bean classes for the synthetic bean archive.
+     *
+     * @param beanClass class to add to the synthetic archive
+     * @return self
+     */
+    Container addBeanClass(Class<?> beanClass);
+
+    /**
+     * All classes from the packages of the specified classes will be added to the set of bean classes for the synthetic bean archive.
+     * <p>
+     * <p>
+     * Note that the scanning possibilities are limited. Therefore, only directories and jar files from the filesystem are supported.
+     * </p>
+     * <p>
+     * <p>
+     * Scanning may also have negative impact on bootstrap performance.
+     * </p>
+     *
+     * @param packageClasses classes whose packages will be added to the synthetic bean archive
+     * @return self
+     */
+    Container packages(Class<?>... packageClasses);
+
+    /**
+     * Packages of the specified classes will be scanned and found classes will be added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param scanRecursively should subpackages be scanned or not
+     * @param packageClasses classes whose packages will be scanned
+     * @return self
+     */
+    Container addPackages(boolean scanRecursively, Class<?>... packageClasses);
+
+    /**
+     * A package of the specified class will be scanned and found classes will be added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param scanRecursively should subpackages be scanned or not
+     * @param packageClass class whose package will be scanned
+     * @return self
+     */
+    Container addPackage(boolean scanRecursively, Class<?> packageClass);
+
+
+    /**
+     * Returns a {@link AnnotatedTypeConfigurator} based on the provided type
+     *
+     * @param type that the {@link AnnotatedType} to configure represents
+     * @param <T> required type
+     * @return an AnnotatedTypeConfigurator whose configuration will be added as new AnnotatedType
+     */
+    <T> AnnotatedTypeConfigurator<T> configureAnnotatedType(Class<T> type);
+
+
+    /**
+     * Add a given set of {@link AnnotatedType} to the set of discovered types.
+     *
+     *
+     * @param annotatedTypes the AnnotatedTypes to add
+     * @return self
+     */
+    Container addAnnotatedTypes(AnnotatedType<?>... annotatedTypes);
+
+
+    /**
+     * Define the set of extensions.
+     *
+     * @param extensions extensions to use in the container
+     * @return self
+     */
+    Container extensions(Extension... extensions);
+
+    /**
+     * Add an extension to the set of extensions.
+     *
+     * @param extension extension to add
+     * @return self
+     */
+    Container addExtension(Extension extension);
```

Comment:
The container should also support adding extension classes, rather than instances.